pkg/bgpv1/agent/controller.go

```diff
@@ -470,3 +484,16 @@ func (c *Controller) validatePolicy(policy *v2alpha1api.CiliumBGPPeeringPolicy)
 	}
 	return nil
 }
+
+// setDefaultPeerPort sets the PeerPort of the provided neighbor
+// to defaultPeerPort (179) if unset.
+func setDefaultPeerPort(n *v2alpha1api.CiliumBGPNeighbor) {
+	if n.PeerPort == nil {
+		n.PeerPort = PtrPort(types.DefaultPeerPort)
+	}
+}
+
+// PtrPort returns the provided port as a pointer.
+func PtrPort(port uint16) *uint16 {
+	return &port
+}
```

Comment:
@YutaroHayakawa the field is a pointer because it's optional.

Reply:
I may be missing something, but it's not mandatory that make the optional field a pointer. For example, `CiliumBGPVirtualRouter.ExportPodCIDR` is optional, but it's not a pointer.

Reply:
@YutaroHayakawa thanks again for the review. I am following the [k8s API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required) for adding `PeerPort` as a pointer. Regarding `ExportPodCIDR` field, it appears that `CiliumBGPVirtualRouter` contains some inconsistencies in how optional fields are represented because the `ServiceSelector` field is optional and a pointer. I can change the field type if you have a strong opinion, just LMK.

Reply:
Got it. Then, I think we don't have to change it for now. Indeed, we seem to have inconsistencies in the fields. Thanks for finding that!